### PR TITLE
Adding 8 white-space-instrinsic-size tests

### PR DESCRIPTION
**Pull request by TalbotG**
6 min-content sizing tests and 2 max-content sizing tests:

white-space-intrinsic-size-013.html
white-space-intrinsic-size-014.html
white-space-intrinsic-size-015.html
white-space-intrinsic-size-016.html
white-space-intrinsic-size-017.html
white-space-intrinsic-size-018.html
white-space-intrinsic-size-019.html
white-space-intrinsic-size-020.html
reference/ref-filled-green-100px-square.xht

### CODE REVIEW

File: css/css-text/white-space/white-space-intrinsic-size-013.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  <title>CSS Text Test: min-content sizing and 'white-space: pre-wrap'</title>
+
+  <link rel="author" title="Gérard Talbot" href="http://www.gtalbot.org/BrowserBugsSection/css21testsuite/">
+  <link rel="help" href="https://www.w3.org/TR/css-text-3/#white-space-property">
+  <link rel="help" href="https://www.w3.org/TR/css-text-3/#white-space-phase-2">
+  <link rel="match" href="reference/ref-filled-green-100px-square.xht">
+  <link rel="stylesheet" type="text/css" href="/fonts/ahem.css">
+
+  <meta content="" name="flags">
+  <meta content="When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line affect the intrinsic min-content size." name="assert">
```
Comment (frivoal) [inline css/css-text/white-space/white-space-intrinsic-size-013.html]: >When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line affect the intrinsic min-content size

That's not correct.
When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line are indeed preserved (and so can be shown if you underline them, but a span around them with a background, etc).
However, according to [4.2](https://drafts.csswg.org/css-text-3/#white-space-phase-2) point 4 bullet 2, they must **hang** (those at the end of wrapped lines hang unconditionally, those before forced breaks hang conditionally, but that difference is not relevant in this test). If you look at the [definition of hanging in 8.2](https://drafts.csswg.org/css-text-3/#hanging), it says:
> The hanging glyph is also not taken into account when computing intrinsic sizes (min-content size and max-content size)

and

> Glyphs that conditionally hang are not taken into account when computing min-content sizes and any sizes derived thereof

So the premise of this test is mistaken, and the green box should be 25px wide (1 letter). Several current implementations are incorrect, so don't trust the results you get in Firefox and Chrome. If I am not mistaken, you'll get correct results for min-content sizing in webkit (not blink) only at the moment.

File: css/css-text/white-space/white-space-intrinsic-size-014.html [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  <title>CSS Text Test: min-content sizing and 'white-space: pre-wrap'</title>
+
+  <link rel="author" title="Gérard Talbot" href="http://www.gtalbot.org/BrowserBugsSection/css21testsuite/">
+  <link rel="help" href="https://www.w3.org/TR/css-text-3/#white-space-property">
+  <link rel="help" href="https://www.w3.org/TR/css-text-3/#white-space-phase-2">
+  <link rel="match" href="reference/ref-filled-green-100px-square.xht">
+  <link rel="stylesheet" type="text/css" href="/fonts/ahem.css">
+
+  <meta content="" name="flags">
+  <meta content="When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line affect the intrinsic min-content size. Overflow-wrap makes no difference." name="assert">
```
Comment (frivoal) [inline css/css-text/white-space/white-space-intrinsic-size-014.html]: same comment as for -013

**Review by frivoal — CHANGES_REQUESTED** [review]
Two tests (-013 and -014) need corrections. The rest is good.

File: css/css-text/white-space/white-space-intrinsic-size-013.html (hunk 1 above)
Comment (TalbotG) [reply css/css-text/white-space/white-space-intrinsic-size-013.html]: Thank you for your complete reply. I have redesigned entirely these 2 tests.

**Comment by TalbotG** [comment]
I think I now understand the conditionally versus unconditionally 'hang'.
I redesigned completely both tests. I am confident about those 2 tests now.

**Review by frivoal — APPROVED** [review]
This is all good now, approved. There was a small mistake in the assertion description, but the behavior of the test is not affected. Please see https://github.com/web-platform-tests/wpt/pull/21967/commits/9884a20b3dbd4b2e33a3581c9388533754115cf0 for details.

**Pull request merged**